src/app/weather/weather.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core';
+import { WeatherService } from '../weather.service';
+import { HttpClientModule } from '@angular/common/http';
+import { PLATFORM_ID, Inject } from '@angular/core';
+import { isPlatformBrowser } from '@angular/common';
+@Component({
+  selector: 'app-weather',
+  standalone: true,
+  imports: [HttpClientModule],
+  templateUrl: './weather.component.html',
+  styleUrl: './weather.component.css'
+})
+export class WeatherComponent implements OnInit {
+  currentCity: string = '';
+  currentTemperature: number = 0;
+  currentHumidity: number = 0;
+  currentWindSpeed: number = 0;
+  currentWeatherCondition: string = '';
+
+  constructor(
+    private weatherService: WeatherService,
+    @Inject(PLATFORM_ID) private platformId: Object
+  ) { }
+
+  ngOnInit(): void {
+    if (isPlatformBrowser(this.platformId)) {
+      navigator.geolocation.getCurrentPosition(position => {
+        const lat = position.coords.latitude;
+        const lon = position.coords.longitude;
+
+        this.weatherService.getCurrentWeather(lat, lon).subscribe((data: any) => {
```

Comment:
1) try to avoid any, and create an interface for response.
2) create a separate service for preparing responses in the model.
3) If using subscribe need to unsubscribe (https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f) or use async pipe 
https://www.telerik.com/blogs/angular-basics-step-by-step-understanding-async-pipe